14_functions_lab/05.py

```diff
@@ -1,10 +1,12 @@
-"""
-Write a function groupby that takes a list
-and a function and returns a dictionary
-keyd by the return value of the function on the list items
-
-For example:
-    groupby(lambda s: s[0], ['foo', 'fi', 'hello', 'hi'])
-    returns: { 'f': ['foo','fi'], 'h': ['hello', 'hi'] }
-"""
-
+def GroupBy(func, myList):
+    myDict = {}
+    for el in myList:
+        theKey =  func(el)
+        if myDict.get(theKey) == None:

```

Comment:
get has a second argument, which is the value to return if key is not found. So better to use: `myDict.get(theKey, []).append(el)`